src/main/java/alexiil/mc/mod/pipes/client/model/SimpleBakedModel.java

```diff
@@ -96,14 +96,14 @@ private static ModelTransformation scale(ModelTransformation from, double by) {
     private static Transformation scale(Transformation from, double by) {
 
         float scale = (float) by;
-        Vec3f nScale = from.scale.copy();
-        nScale.scale(scale);
+        Vector3f nScale = from.scale;
+        nScale.mul(scale);
```

Comment:
This mutation will cause some issues, as no copies are performed here.

Reply:
I guess you could do something like `from.scale.mul(scale, new Vector3f())` which performs the multiplication, storing it into the new vector. And then you would just remove the lower `mul(scale)`.